components/mediation/data-publishers/org.wso2.micro.integrator.prometheus.publisher/src/main/java/org/wso2/micro/integrator/prometheus/publisher/service/MetricResource.java

```diff
@@ -39,12 +43,15 @@
 
     private static Log log = LogFactory.getLog(MetricResource.class);
     private MetricPublisher metricPublisher;
+    private MetricFormatter metricFormatter;
     public static final String NO_ENTITY_BODY = "NO_ENTITY_BODY";
+    private CollectorRegistry registry = CollectorRegistry.defaultRegistry;
 
     public MetricResource(String urlTemplate) {
 
         super(urlTemplate);
         metricPublisher = new MetricPublisher();
+        metricFormatter = new MetricFormatter();
```

Comment:
If we make MetricFormatter as a static util class, we don't need this here.

Reply:
Made the method static and removed initialization.